beater/handlers.go

```diff
@@ -52,17 +52,70 @@ type routeMapping struct {
 	ProcessorFactory
 }
 
+type serverResponse struct {
+	err     error
+	code    int
+	counter *monitoring.Int
+}
+
 var (
-	serverMetrics  = monitoring.Default.NewRegistry("apm-server.server")
-	requestCounter = monitoring.NewInt(serverMetrics, "requests.counter")
-	responseValid  = monitoring.NewInt(serverMetrics, "response.valid")
-	responseErrors = monitoring.NewInt(serverMetrics, "response.errors")
+	serverMetrics = monitoring.Default.NewRegistry("apm-server.server")
+	counter       = func(s string) *monitoring.Int {
+		return monitoring.NewInt(serverMetrics, s)
+	}
+	requestCounter   = counter("request.count")
+	responseCounter  = counter("response.count")
+	responseErrors   = counter("response.errors.count")
+	responseSucesses = counter("response.valid.count")
 
-	errInvalidToken            = errors.New("invalid token")
-	errForbidden               = errors.New("forbidden request")
-	errPOSTRequestOnly         = errors.New("only POST requests are supported")
-	errTooManyRequests         = errors.New("too many requests")
-	errConcurrencyLimitReached = errors.New("request timed out waiting to be processed. Increase `concurrent_requests` or `max_request_queue_time` or reduce payload size")
+	okResponse = serverResponse{
+		nil, http.StatusOK, counter("response.valid.okResponse"),
+	}
+	acceptedResponse = serverResponse{
+		nil, http.StatusAccepted, counter("response.valid.acceptedResponse"),
+	}
+	forbiddenResponse = serverResponse{
+		errors.New("forbiddenResponse request"), http.StatusForbidden, counter("response.errors.forbiddenResponse"),
+	}
+	unauthorizedResponse = serverResponse{
+		errors.New("invalid token"), http.StatusUnauthorized, counter("response.errors.unauthorizedResponse"),
+	}
+	requestTooLargeResponse = serverResponse{
+		errors.New("request body too large"), http.StatusRequestEntityTooLarge, counter("response.errors.toolarge"),
```

Comment:
this counter is called `response.errors.toolarge`, but the one above is `response.errors.forbiddenResponse`. We should make it consistent whether we include `Response` postfix. I'm fine with either of the two